bin/run.sh

```diff
@@ -24,36 +24,45 @@ fi
 slug="$1"
 input_dir="${2%/}"
 output_dir="${3%/}"
+
 exercise=$(echo "${slug}" | sed -E 's/(^|-)([a-z])/\U\2/g')
-tests_file="${input_dir}/src/test/scala/${exercise}Test.scala"
-tests_results_file="${input_dir}/target/test-reports/TEST-${exercise}Test.xml"
-original_tests_file="${input_dir}/src/test/scala/${exercise}Test.scala.original"
+
+test_runner_jar=/opt/test-runner/target/scala-2.13/TestRunner-assembly-0.1.0-SNAPSHOT.jar
+
+workdir=/tmp/exercise
+workdir_target=/tmp/exercise/target
+
 results_file="${output_dir}/results.json"
 build_log_file="${output_dir}/build.log"
 runner_log_file="${output_dir}/runner.log"
+tests_results_file="${workdir}/test-reports/TEST-${exercise}Test.xml"
+
+# ensure scala binaries are available in the PATH
+export PATH="$PATH:/opt/test-runner/scala-2.13.8/bin"
 
 # Create the output directory if it doesn't exist
 mkdir -p "${output_dir}"
 
-echo "${slug}: testing..."
+# ensure a clean workdir
+rm -rf "${workdir}"
+mkdir -p "${workdir}"
+mkdir -p "${workdir_target}"
 
-pushd "${input_dir}" > /dev/null
+echo
+echo "${slug}: testing..."
 
-cp "${tests_file}" "${original_tests_file}"
+cp -R "${input_dir}"/src "${workdir}"
 
 # Enable all pending tests
-sed -i 's/pending//g' "${tests_file}"
-
-sbt "set offline := true" test > "${build_log_file}"
-
-# Sanitize the output
-sed -i -E '/^\[info\]/d' "${build_log_file}"
+sed -i 's/pending//g' "${workdir}"/src/test/scala/*
 
-mv -f "${original_tests_file}" "${tests_file}"
+# compile source and tests
+scalac -classpath "${test_runner_jar}" -d "${workdir_target}" "${workdir}"/src/main/scala/* "${workdir}"/src/test/scala/* &> "${build_log_file}"
```

Comment:
Those will only work if those libraries are added to the `scala-test-runner` dependencies, so that they are added to the fat jar built in the Dockerfile with `sbt assembly`. In other words, we can make it work :smile: 

Does that work with the current code? Given the fact that the docker images are run with network disabled, I assume not, but maybe I missed some piece of the puzzle.

Reply:
> Those will only work if those libraries are added to the scala-test-runner dependencies, so that they are added to the fat jar built in the Dockerfile with sbt assembly. In other words, we can make it work 😄

Great! I've just ran a little bash command to try and see which libraries are being used, and this is what I found:

"com.github.julien-truffaut" %%  "monocle-core"  % "2.0.0",
"com.github.julien-truffaut" %%  "monocle-macro" % "2.0.0"
"org.scalaz" %% "scalaz-core" % "7.2.28"
"org.scala-lang.modules" %% "scala-parser-combinators" % "2.1.0"
"org.scalacheck" %% "scalacheck" % "1.14.1" % "test"
"org.scalatestplus" %% "scalacheck-1-15" % "3.2.9.0" % "test" 

If we could ensure that all these libraries work in the test runner, we should be good for now. Would you be interested in doing a follow-up PR to add this? If so, I'd suggest adding new test cases to the `tests` dir to verify that they can use these libraries successfully.

Reply:
Yes, I can do that. Can you point me to the specific tests we need to support? I would like to have a look at the code and also understand why those libraries are needed in the first place :smile: